packages/bun-uws/src/AsyncSocket.h

```diff
@@ -166,33 +166,34 @@ struct AsyncSocket {
         LoopData *loopData = getLoopData();
         BackPressure &backPressure = getAsyncSocketData()->buffer;
         size_t existingBackpressure = backPressure.length();
-        if ((!existingBackpressure) && (isCorked() || canCork()) && (loopData->corkOffset + size < LoopData::CORK_BUFFER_SIZE)) {
+        if ((!existingBackpressure) && (isCorked() || canCork()) && (loopData->getCorkOffset() + size < LoopData::CORK_BUFFER_SIZE)) {
             /* Cork automatically if we can */
             if (isCorked()) {
-                char *sendBuffer = loopData->corkBuffer + loopData->corkOffset;
-                loopData->corkOffset += (unsigned int) size;
+                char *sendBuffer = loopData->getCorkSendBuffer();
+                loopData->incrementCorkedOffset((unsigned int) size);
                 return {sendBuffer, SendBufferAttribute::NEEDS_NOTHING};
             } else {
                 cork();
-                char *sendBuffer = loopData->corkBuffer + loopData->corkOffset;
-                loopData->corkOffset += (unsigned int) size;
+                char *sendBuffer = loopData->getCorkSendBuffer();
+                loopData->incrementCorkedOffset((unsigned int) size);
                 return {sendBuffer, SendBufferAttribute::NEEDS_UNCORK};
             }
         } else {
 
             /* If we are corked and there is already data in the cork buffer,
             mark how much is ours and reset it */
             unsigned int ourCorkOffset = 0;
-            if (isCorked() && loopData->corkOffset) {
-                ourCorkOffset = loopData->corkOffset;
-                loopData->corkOffset = 0;
+
+            if (isCorked()) {
+                ourCorkOffset = loopData->getCorkOffset();
+                loopData->setCorkOffset(0);
             }
 
             /* Fallback is to use the backpressure as buffer */
             backPressure.resize(ourCorkOffset + existingBackpressure + size);
 
             /* And copy corkbuffer in front */
-            memcpy((char *) backPressure.data() + existingBackpressure, loopData->corkBuffer, ourCorkOffset);
+            memcpy((char *) backPressure.data() + existingBackpressure, loopData->getCorkBuffer(), ourCorkOffset);
```

Comment:
`ourCorkOffset` will never be zero right?

Reply:
I think it can be in edge cases, still worth checking